mllib/src/main/scala/org/apache/spark/mllib/clustering/LocalKMeans.scala

```diff
@@ -66,6 +69,12 @@ private[mllib] object LocalKMeans extends Logging {
       } else {
         centers(i) = points(j - 1).toDense
       }
+
+      //update costArray
+      for(p <- 0 to points.length-1){
+        costArray(p)=math.min(KMeans.fastSquaredDistance(points(p), centers(i)),costArray(p))
```

Comment:
JIRA already created and linked, 
I will fix this once I have access to my work machine